bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingTypeResource.java

```diff
@@ -101,32 +96,29 @@ public Response getByUID(@PathParam("thingTypeUID") String thingTypeUID,
-                    ConfigDescriptionParameterBean configDescriptionParameterBean = new ConfigDescriptionParameterBean(
-                            configDescriptionParameter.getName(), configDescriptionParameter.getType(),
-                            configDescriptionParameter.getMinimum(), configDescriptionParameter.getMaximum(),
-                            configDescriptionParameter.getStepSize(), configDescriptionParameter.getPattern(),
-                            configDescriptionParameter.isRequired(), configDescriptionParameter.isReadOnly(),
-                            configDescriptionParameter.isMultiple(), configDescriptionParameter.getContext(),
-                            String.valueOf(configDescriptionParameter.getDefault()),
-                            configDescriptionParameter.getLabel(), configDescriptionParameter.getDescription(),
-                            createBeansForOptions(configDescriptionParameter.getOptions()),
-                            createBeansForCriteria(configDescriptionParameter.getFilterCriteria()));
-                    configDescriptionParameterBeans.add(configDescriptionParameterBean);
-                }
-                return configDescriptionParameterBeans;
+
+        ConfigDescription configDescription = configDescriptionRegistry.getConfigDescription(configDescriptionURI, locale);
+        if (configDescription != null) {
+            List<ConfigDescriptionParameterBean> configDescriptionParameterBeans = new ArrayList<>(
+                    configDescription.getParameters().size());
+            for (ConfigDescriptionParameter configDescriptionParameter : configDescription.getParameters()) {
+                ConfigDescriptionParameterBean configDescriptionParameterBean = new ConfigDescriptionParameterBean(
+                        configDescriptionParameter.getName(), configDescriptionParameter.getType(),
+                        configDescriptionParameter.getMinimum(), configDescriptionParameter.getMaximum(),
+                        configDescriptionParameter.getStepSize(), configDescriptionParameter.getPattern(),
+                        configDescriptionParameter.isRequired(), configDescriptionParameter.isReadOnly(),
+                        configDescriptionParameter.isMultiple(), configDescriptionParameter.getContext(),
+                        String.valueOf(configDescriptionParameter.getDefault()),
+                        configDescriptionParameter.getLabel(), configDescriptionParameter.getDescription(),
+                        createBeansForOptions(configDescriptionParameter.getOptions()),
+                        createBeansForCriteria(configDescriptionParameter.getFilterCriteria()));
+                configDescriptionParameterBeans.add(configDescriptionParameterBean);
             }
-        } catch (URISyntaxException ex) {
-            logger.error(ex.getMessage(), ex);
+            return configDescriptionParameterBeans;
         }
+       
```

Comment:
No, the formatter does clean up the code..
It would be better to use the "clean up" function instead of the formatter. The ESH clean up rule applies the formatter and some more stuff (e.g. remove trailing whitespaces).

The Eclipse IDE also support to add this stuff as "save actions". IMHO Kai will add this to the Yoxos profile in the next days.